src/lib/tests/Generator.class.test.ts

```diff
@@ -24,26 +24,28 @@ describe('Generator Class', () => {
                             query: 'file://static/example/iterator-stage-1.rq',
                             endpoint: 'file://static/tests/iris.nt'
                         },
-                        generator: {
-                            query: 'file://static/example/generator-stage-1.rq'
+                        generator: [
+{
+                            query: 'file://static/example/generator-stage-1-1.rq'
                         }
-                    },
+]                    },
                     {
                         name: 'Stage 2',
                         iterator: {
                             query: 'file://static/example/iterator-stage-2.rq',
                         },
-                        generator: {
+                        generator: [
+{
                             query: 'file://static/example/generator-stage-2.rq',
                             endpoint: 'file://static/tests/wikidata.nt'
                         }
-                    }
+]                    }
                 ]
             }
             const pipeline = new Pipeline(configuration, {silent: true})
             const stageConfig = configuration.stages[0]
             const stage = new Stage(pipeline, stageConfig)
-            const generator = new Generator(stage)
+            const generator = new Generator(stage, 0)
```

Comment:
What does the `0` mean?

Or more general: if I can't  easily determine what the `0` means here, the arguments of the constructor of `Generator` should maybe be changed to an options object, s.t. the caller names the arguments (and the reader of the calling code can then read the names).

Reply:
`0` in this case is the index of the generator, given the stage. A stage can for example have 2 generators, this index specifies which one of the generators it is, and which configuration from this stage the Generator class should use.